src/protocols/ip6.rs

```diff
@@ -0,0 +1,75 @@
+//! IPv6 Protocol
+//!
+//! It implements a simple packet-oriented interface that can be used by
+//! drivers, daemons, and applications to transmit and receive network packets.
+
+use crate::efi::{Boolean, Ipv6Address, MacAddress};
+
+#[repr(C)]
+pub struct ConfigData {
+    default_protocol: u8,
+    accept_any_protocol: Boolean,
+    accept_icmp_errors: Boolean,
+    accept_promiscuous: Boolean,
+    destination_address: Ipv6Address,
+    station_address: Ipv6Address,
+    traffic_class: u8,
+    hop_limit: u8,
+    flow_lable: u32,
+    receive_timeout: u32,
+    transmit_timeout: u32,
+}
+
+#[repr(C)]
+pub struct AddressInfo {
+    address: Ipv6Address,
+    prefix_length: u8,
+}
+
+#[repr(C)]
+pub struct RouteTable {
+    gateway: Ipv6Address,
+    destination: Ipv6Address,
+    prefix_length: u8,
+}
+
+pub type NeighboutState = u32;
```

Comment:
`Neighbout` -> `Neighbor`

Also note that American-English spells it with `o` instead of `ou` (as in British English). The spec uses the American style, so I fixed it all over the place.